docs/how-tos/sensor-module.md

```diff
@@ -0,0 +1,464 @@
+
+```python {class="line-numbers linkable-line-numbers"}
+# test-script.py
+import openmeteo_requests
+import requests_cache
+from retry_requests import retry
+
+# Set up the Open-Meteo API client with cache and retry on error
+cache_session = requests_cache.CachedSession('.cache', expire_after=3600)
+retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+openmeteo = openmeteo_requests.Client(session=retry_session)
+
+# Make sure all required weather variables are listed here
+# The order of variables in hourly or daily is important
+# to assign them correctly below
+url = "https://air-quality-api.open-meteo.com/v1/air-quality"
+params = {
+  "latitude": 44.0582,
+  "longitude": -121.3153,
+  "current": ["pm10", "pm2_5"],
+  "timezone": "America/Los_Angeles"
+}
+responses = openmeteo.weather_api(url, params=params)
+
+# Process first location.
+response = responses[0]
+print(f"Coordinates {response.Latitude()}°N {response.Longitude()}°E")
+print(f"Elevation {response.Elevation()} m asl")
+print(f"Timezone {response.Timezone()} {response.TimezoneAbbreviation()}")
+print(f"Timezone difference to GMT+0 {response.UtcOffsetSeconds()} s")
+
+# Current values. The order of variables needs to be the same as requested.
+current = response.Current()
+current_pm10 = current.Variables(0).Value()
+current_pm2_5 = current.Variables(1).Value()
+
+print(f"Current time {current.Time()}")
+print(f"Current pm10 {current_pm10}")
+print(f"Current pm2_5 {current_pm2_5}")
+```
+
+{{< /expand >}}
+
+{{< expand "An example of getting PM2.5 and PM10 readings over serial from sensor hardware" >}}
+This script is based on the [pms7003 module](https://app.viam.com/module/joyce/pms7003).
+Before creating the module, the following script could have been used to verify the code that gets the readings from the sensor.
+
+```python {class="line-numbers linkable-line-numbers"}
+# my-sensor-test.py
+from serial import Serial
+import time
+
+
+def main():
+
+    port = Serial('/dev/ttyAMA0', baudrate=9600)
+
+    def parse_data(data):
+
+        if len(data) < 2:
+            return {}
+        if data[0] == 0x42 and data[1] == 0x4d:
+            pm2_5_atm = (data[12] << 8) + data[13]
+            pm10_atm = (data[14] << 8) + data[15]
+            print(f'PM2.5 (atmospheric): {pm2_5_atm} µg/m3')
+            print(f'PM10 (atmospheric): {pm10_atm} µg/m3')
+
+            # Return a dictionary of the readings
+            return {
+                "pm2_5_atm": pm2_5_atm,
+                "pm10_atm": pm10_atm,
+            }
+
+        else:
+            print('Data does not start with the expected start bytes.')
+            return {}
+
+    while port.in_waiting == 0:
+        time.sleep(0.01)  # wait for 10 ms
+
+    data = port.read(port.in_waiting)
+    print(parse_data(data))
+
+
+if __name__ == "__main__":
+    main()
+```
+
+{{< /expand >}}
+{{% /tablestep %}}
+{{% tablestep %}}
+**2. Run your test script**
```

Comment:
You can if you want to skip having steps for this part. I agree that here it feels a bit unnecessary. In the rest it's more helpful. Up to you.

Reply:
Sorry i'm not sure what this comment is in relation to

Reply:
I'm saying you don't need to put tablesteps here into the "Start with a test script" section if you don't want to

Reply:
oooh